File: frontend/src/utils/comprehensivePermissions.ts
/**
 * Comprehensive Permissions System
 * Includes all menus, child menus, and screening forms for RBAC role creation
 */

export interface Permission {
  id: string;
  name: string;
  description: string;
  category: string;
  resource: string;
  action: string;
}

/**
 * Complete permission set for all system features
 */
export const COMPREHENSIVE_PERMISSIONS: Permission[] = [
  // ========== DASHBOARD PERMISSIONS ==========
  {
    id: 'dashboard_view',
    name: 'View Dashboard',
    description: 'Access to main dashboard',
    category: 'dashboard',
    resource: 'dashboard',
    action: 'view'
  },
  {
    id: 'analytics_view',
    name: 'View Health Analytics',
    description: 'Access to health analytics and insights',
    category: 'dashboard',
    resource: 'analytics',
    action: 'view'
  },

  // ========== USER MANAGEMENT PERMISSIONS ==========
  {
    id: 'users_view',
    name: 'View Users',
    description: 'View user directory and information',
    category: 'user',
    resource: 'users',
    action: 'view'
  },
  {
    id: 'users_create',
    name: 'Create Users',
    description: 'Create new user accounts',
    category: 'user',
    resource: 'users',
    action: 'create'
  },
  {
    id: 'users_update',
    name: 'Update Users',
    description: 'Edit existing user accounts',
    category: 'user',
    resource: 'users',
    action: 'update'
  },
  {
    id: 'users_delete',
    name: 'Delete Users',
    description: 'Remove user accounts',
    category: 'user',
    resource: 'users',
    action: 'delete'
  },
  {
    id: 'users_manage',
    name: 'Manage Users',
    description: 'Full user management capabilities',
    category: 'user',
    resource: 'users',
    action: 'manage'
  },

  // ========== MEDICAL STAFF PERMISSIONS ==========
  {
    id: 'medical_staff_view',
    name: 'View Medical Staff',
    description: 'View medical staff directory',
    category: 'medical',
    resource: 'medical_staff',
    action: 'view'
  },
  {
    id: 'medical_staff_manage',
    name: 'Manage Medical Staff',
    description: 'Full medical staff management',
    category: 'medical',
    resource: 'medical_staff',
    action: 'manage'
  },

  // ========== PATIENT MANAGEMENT PERMISSIONS ==========
  {
    id: 'patients_view',
    name: 'View Patients',
    description: 'View patient records and information',
    category: 'patient',
    resource: 'patients',
    action: 'view'
  },
  {
    id: 'patients_create',
    name: 'Create Patients',
    description: 'Register new patients',
    category: 'patient',
    resource: 'patients',
    action: 'create'
  },
  {
    id: 'patients_update',
    name: 'Update Patients',
    description: 'Edit patient information',
    category: 'patient',
    resource: 'patients',
    action: 'update'
  },
  {
    id: 'patients_delete',
    name: 'Delete Patients',
    description: 'Remove patient records',
    category: 'patient',
    resource: 'patients',
    action: 'delete'
  },

  // ========== SCHOOL MANAGEMENT PERMISSIONS ==========
  {
    id: 'schools_view',
    name: 'View Schools',
    description: 'View school information',
    category: 'school',
    resource: 'schools',
    action: 'view'
  },
  {
    id: 'schools_manage',
    name: 'Manage Schools',
    description: 'Full school management',
    category: 'school',
    resource: 'schools',
    action: 'manage'
  },
  {
    id: 'students_view',
    name: 'View Students',
    description: 'View student records',
    category: 'school',
    resource: 'students',
    action: 'view'
  },
  {
    id: 'students_manage',
    name: 'Manage Students',
    description: 'Full student management',
    category: 'school',
    resource: 'students',
    action: 'manage'
  },
  {
    id: 'teachers_view',
    name: 'View Teachers',
    description: 'View teacher information',
    category: 'school',
    resource: 'teachers',
    action: 'view'
  },
  {
    id: 'teachers_manage',
    name: 'Manage Teachers',
    description: 'Full teacher management',
    category: 'school',
    resource: 'teachers',
    action: 'manage'
  },
  {
    id: 'parents_view',
    name: 'View Parents',
    description: 'View parent information',
    category: 'school',
    resource: 'parents',
    action: 'view'
  },
  {
    id: 'parents_manage',
    name: 'Manage Parents',
    description: 'Full parent management',
    category: 'school',
    resource: 'parents',
    action: 'manage'
  },

  // ========== SCREENING FORM PERMISSIONS ==========
  {
    id: 'screening_mobile_access',
    name: 'Mobile Vision Screening',
    description: 'Access to mobile vision screening form',
    category: 'screening',
    resource: 'screening_forms',
    action: 'mobile_vision'
  },
  {
    id: 'screening_standard_access',
    name: 'Standard Vision Screening',
    description: 'Access to standard vision screening form',
    category: 'screening',
    resource: 'screening_forms',
    action: 'standard_vision'
  },
  {
    id: 'screening_enhanced_access',
    name: 'Enhanced Screening Interface',
    description: 'Access to enhanced screening tools',
    category: 'screening',
    resource: 'screening_forms',
    action: 'enhanced_interface'
  },
  {
    id: 'screening_va_access',
    name: 'VA Screening Interface',
    description: 'Access to visual acuity screening interface',
    category: 'screening',
    resource: 'screening_forms',
    action: 'va_interface'
  },
  {
    id: 'screening_school_access',
    name: 'School Screening Form',
    description: 'Access to school-based screening form',
    category: 'screening',
    resource: 'screening_forms',
    action: 'school_screening'
  },
  {
    id: 'screening_outcome_access',
    name: 'Screening Outcome Form',
    description: 'Access to screening outcome documentation',
    category: 'screening',
    resource: 'screening_forms',
    action: 'outcome_form'
  },
  {
    id: 'screening_diagnosis_access',
    name: 'Doctor Diagnosis Form',
    description: 'Access to medical diagnosis form',
    category: 'screening',
    resource: 'screening_forms',
    action: 'doctor_diagnosis'
  },

  // ========== SCREENING TYPE PERMISSIONS ==========
  {
    id: 'screening_type_basic',
    name: 'Basic Screening Types',
    description: 'Access to basic screening types (visual acuity, distance, near)',
    category: 'screening',
    resource: 'screening_types',
    action: 'basic'
  },
  {
    id: 'screening_type_advanced',
    name: 'Advanced Screening Types',
    description: 'Access to advanced screening types (comprehensive, color, depth)',
    category: 'screening',
    resource: 'screening_types',
    action: 'advanced'
  },
  {
    id: 'screening_type_specialized',
    name: 'Specialized Screening Types',
    description: 'Access to specialized screening protocols',
    category: 'screening',
    resource: 'screening_types',
    action: 'specialized'
  },
  {
    id: 'screening_type_diagnostic',
    name: 'Diagnostic Screening Types',
    description: 'Access to diagnostic screening tools',
    category: 'screening',
    resource: 'screening_types',
    action: 'diagnostic'
  },

  // ========== SCREENING MANAGEMENT PERMISSIONS ==========
  {
    id: 'screenings_view',
    name: 'View Screenings',
    description: 'View screening sessions and results',
    category: 'screening',
    resource: 'screenings',
    action: 'view'
  },
  {
    id: 'screenings_create',
    name: 'Create Screenings',
    description: 'Start new screening sessions',
    category: 'screening',
    resource: 'screenings',
    action: 'create'
  },
  {
    id: 'screenings_update',
    name: 'Update Screenings',
    description: 'Edit screening sessions and results',
    category: 'screening',
    resource: 'screenings',
    action: 'update'
  },
  {
    id: 'screenings_delete',
    name: 'Delete Screenings',
    description: 'Remove screening sessions',
    category: 'screening',
    resource: 'screenings',
    action: 'delete'
  },
  {
    id: 'screenings_manage',
    name: 'Manage Screenings',
    description: 'Full screening management capabilities',
    category: 'screening',
    resource: 'screenings',
    action: 'manage'
  },

  // ========== MEDICAL REPORTS PERMISSIONS ==========
  {
    id: 'reports_view',
    name: 'View Medical Reports',
    description: 'View medical reports and analytics',
    category: 'medical',
    resource: 'reports',
    action: 'view'
  },
  {
    id: 'reports_create',
    name: 'Generate Medical Reports',
    description: 'Create and generate medical reports',
    category: 'medical',
    resource: 'reports',
    action: 'create'
  },
  {
    id: 'reports_export',
    name: 'Export Medical Reports',
    description: 'Export reports to various formats',
    category: 'medical',
    resource: 'reports',
    action: 'export'
  },

  // ========== INVENTORY MANAGEMENT PERMISSIONS ==========
  {
    id: 'inventory_view',
    name: 'View Inventory',
    description: 'View glasses and equipment inventory',
    category: 'inventory',
    resource: 'inventory',
    action: 'view'
  },
  {
    id: 'inventory_manage',
    name: 'Manage Inventory',
    description: 'Full inventory management',
    category: 'inventory',
    resource: 'inventory',
    action: 'manage'
  },
  {
    id: 'delivery_manage',
    name: 'Manage Deliveries',
    description: 'Track and manage glasses deliveries',
    category: 'inventory',
    resource: 'delivery',
    action: 'manage'
  },

  // ========== SYSTEM ADMINISTRATION PERMISSIONS ==========
  {
    id: 'panel_settings_view',
    name: 'View Panel Settings',
    description: 'View system panel settings',
    category: 'system',
    resource: 'panel_settings',
    action: 'view'
  },
  {
    id: 'panel_settings_manage',
    name: 'Manage Panel Settings',
    description: 'Configure system panel settings',
    category: 'system',
    resource: 'panel_settings',
    action: 'manage'
  },
  {
    id: 'rbac_view',
    name: 'View RBAC',
    description: 'View roles, permissions, and user assignments',
    category: 'system',
    resource: 'rbac',
    action: 'view'
  },
  {
    id: 'rbac_manage',
    name: 'Manage RBAC',
    description: 'Full RBAC management capabilities',
    category: 'system',
    resource: 'rbac',
    action: 'manage'
  },
  {
    id: 'security_audit',
    name: 'Security Audit',
    description: 'Access to security audit and monitoring',
    category: 'system',
    resource: 'security',
    action: 'audit'
  },

  // ========== AI INSIGHTS PERMISSIONS ==========
  {
    id: 'ai_insights_view',
    name: 'View AI Insights',
    description: 'Access to AI-powered insights and analytics',
    category: 'ai',
    resource: 'ai_insights',
    action: 'view'
  },
  {
    id: 'ai_insights_generate',
    name: 'Generate AI Insights',
    description: 'Generate AI-powered analysis and reports',
    category: 'ai',
    resource: 'ai_insights',
    action: 'generate'
  },

  // ========== COMMUNICATION PERMISSIONS ==========
  {
    id: 'line_notifications_view',
    name: 'View LINE Notifications',
    description: 'View LINE notification settings',
    category: 'communication',
    resource: 'line_notifications',
    action: 'view'
  },
  {
    id: 'line_notifications_manage',
    name: 'Manage LINE Notifications',
    description: 'Configure LINE notification system',
    category: 'communication',
    resource: 'line_notifications',
    action: 'manage'
  },

  // ========== APPOINTMENT PERMISSIONS ==========
  {
    id: 'appointments_view',
    name: 'View Appointments',
    description: 'View appointment schedules',
    category: 'scheduling',
    resource: 'appointments',
    action: 'view'
  },
  {
    id: 'appointments_create',
    name: 'Create Appointments',
    description: 'Schedule new appointments',
    category: 'scheduling',
    resource: 'appointments',
    action: 'create'
  },
  {
    id: 'appointments_manage',
    name: 'Manage Appointments',
    description: 'Full appointment management',
    category: 'scheduling',
    resource: 'appointments',
    action: 'manage'
  },

  // ========== MENU ACCESS PERMISSIONS ==========
  {
    id: 'menu_school_management',
    name: 'School Management Menu',
    description: 'Access to school management menu and features',
    category: 'menu',
    resource: 'school_menu',
    action: 'access'
  },
  {
    id: 'menu_medical_screening',
    name: 'Medical Screening Menu',
    description: 'Access to medical screening menu and features',
    category: 'menu',
    resource: 'medical_menu',
    action: 'access'
  },
  {
    id: 'menu_user_management',
    name: 'User Management Menu',
    description: 'Access to user management menu',
    category: 'menu',
    resource: 'user_menu',
    action: 'access'
  },
  {
    id: 'menu_medical_staff',
    name: 'Medical Staff Menu',
    description: 'Access to medical staff menu',
    category: 'menu',
    resource: 'staff_menu',
    action: 'access'
  },
  {
    id: 'menu_inventory',
    name: 'Inventory Menu',
    description: 'Access to inventory management menu',
    category: 'menu',
    resource: 'inventory_menu',
    action: 'access'
  },
  {
    id: 'menu_panel_settings',
    name: 'Panel Settings Menu',
    description: 'Access to panel settings menu',
    category: 'menu',
    resource: 'settings_menu',
    action: 'access'
  },

  // ========== CHILD MENU PERMISSIONS ==========
  {
    id: 'submenu_students',
    name: 'Students Submenu',
    description: 'Access to students management submenu',
    category: 'submenu',
    resource: 'students_submenu',
    action: 'access'
  },
  {
    id: 'submenu_parents',
    name: 'Parents Submenu',
    description: 'Access to parents management submenu',
    category: 'submenu',
    resource: 'parents_submenu',
    action: 'access'
  },
  {
    id: 'submenu_teachers',
    name: 'Teachers Submenu',
    description: 'Access to teachers management submenu',
    category: 'submenu',
    resource: 'teachers_submenu',
    action: 'access'
  },
  {
    id: 'submenu_school_screenings',
    name: 'School Screenings Submenu',
    description: 'Access to school-based screenings submenu',
    category: 'submenu',
    resource: 'school_screenings_submenu',
    action: 'access'
  },
  {
    id: 'submenu_patient_registration',
    name: 'Patient Registration Submenu',
    description: 'Access to patient registration submenu',
    category: 'submenu',
    resource: 'patient_reg_submenu',
    action: 'access'
  },
  {
    id: 'submenu_va_screening',
    name: 'VA Screening Submenu',
    description: 'Access to VA screening submenu',
    category: 'submenu',
    resource: 'va_screening_submenu',
    action: 'access'
  },
  {
    id: 'submenu_diagnosis',
    name: 'Diagnosis Submenu',
    description: 'Access to diagnosis and treatment submenu',
    category: 'submenu',
    resource: 'diagnosis_submenu',
    action: 'access'
  },
  {
    id: 'submenu_glasses_inventory',
    name: 'Glasses Inventory Submenu',
    description: 'Access to glasses inventory submenu',
    category: 'submenu',
    resource: 'glasses_inventory_submenu',
    action: 'access'
  },
  {
    id: 'submenu_glasses_delivery',
    name: 'Glasses Delivery Submenu',
    description: 'Access to glasses delivery submenu',
    category: 'submenu',
    resource: 'glasses_delivery_submenu',
    action: 'access'
  },
  {
    id: 'submenu_general_settings',
    name: 'General Settings Submenu',
    description: 'Access to general settings submenu',
    category: 'submenu',
    resource: 'general_settings_submenu',
    action: 'access'
  },
  {
    id: 'submenu_rbac_management',
    name: 'RBAC Management Submenu',
    description: 'Access to RBAC management submenu',
    category: 'submenu',
    resource: 'rbac_submenu',
    action: 'access'
  },

  // ========== SPECIAL PERMISSIONS ==========
  {
    id: 'admin_panel_access',
    name: 'Admin Panel Access',
    description: 'Access to system administration panel',
    category: 'system',
    resource: 'admin_panel',
    action: 'access'
  },
  {
    id: 'full_system_access',
    name: 'Full System Access',
    description: 'Complete access to all system features',
    category: 'system',
    resource: 'all',
    action: 'all'
  }
];

/**
 * Get permissions by category
 */
export const getPermissionsByCategory = (category?: string): Permission[] => {
  if (!category) return COMPREHENSIVE_PERMISSIONS;
  return COMPREHENSIVE_PERMISSIONS.filter(p => p.category === category);
};

/**
 * Get all available categories
 */
export const getPermissionCategories = (): string[] => {
  return Array.from(new Set(COMPREHENSIVE_PERMISSIONS.map(p => p.category)));
};

/**
 * Check if permission exists
 */
export const hasPermission = (permissionId: string): boolean => {
  return COMPREHENSIVE_PERMISSIONS.some(p => p.id === permissionId);
};

File: backend/app/utils/comprehensivePermissions.py
# Comprehensive permissions master data for RBAC system
COMPREHENSIVE_PERMISSIONS = [
    # System Administration
    {"id": "system_admin_full", "name": "Full System Administration", "description": "Complete system administration access", "category": "system", "resource": "system", "action": "admin"},
    {"id": "system_settings_manage", "name": "Manage System Settings", "description": "Configure system-wide settings", "category": "system", "resource": "settings", "action": "manage"},
    {"id": "system_logs_view", "name": "View System Logs", "description": "Access system logs and audit trails", "category": "system", "resource": "logs", "action": "view"},
    {"id": "system_backup_manage", "name": "Manage System Backups", "description": "Create and restore system backups", "category": "system", "resource": "backup", "action": "manage"},
    
    # User Management
    {"id": "user_create", "name": "Create Users", "description": "Create new user accounts", "category": "user", "resource": "users", "action": "create"},
    {"id": "user_read", "name": "View Users", "description": "View user account information", "category": "user", "resource": "users", "action": "read"},
    {"id": "user_update", "name": "Update Users", "description": "Modify user account details", "category": "user", "resource": "users", "action": "update"},
    {"id": "user_delete", "name": "Delete Users", "description": "Remove user accounts", "category": "user", "resource": "users", "action": "delete"},
    {"id": "user_roles_assign", "name": "Assign User Roles", "description": "Assign and modify user roles", "category": "user", "resource": "user_roles", "action": "assign"},
    
    # RBAC Management
    {"id": "rbac_roles_create", "name": "Create Roles", "description": "Create new roles in the system", "category": "rbac", "resource": "roles", "action": "create"},
    {"id": "rbac_roles_read", "name": "View Roles", "description": "View role definitions and permissions", "category": "rbac", "resource": "roles", "action": "read"},
    {"id": "rbac_roles_update", "name": "Update Roles", "description": "Modify role definitions and permissions", "category": "rbac", "resource": "roles", "action": "update"},
    {"id": "rbac_roles_delete", "name": "Delete Roles", "description": "Remove roles from the system", "category": "rbac", "resource": "roles", "action": "delete"},
    {"id": "rbac_permissions_manage", "name": "Manage Permissions", "description": "Create and modify system permissions", "category": "rbac", "resource": "permissions", "action": "manage"},
    
    # Medical Staff Management
    {"id": "medical_staff_create", "name": "Create Medical Staff", "description": "Add new medical staff members", "category": "medical", "resource": "staff", "action": "create"},
    {"id": "medical_staff_read", "name": "View Medical Staff", "description": "View medical staff information", "category": "medical", "resource": "staff", "action": "read"},
    {"id": "medical_staff_update", "name": "Update Medical Staff", "description": "Modify medical staff details", "category": "medical", "resource": "staff", "action": "update"},
    {"id": "medical_staff_delete", "name": "Delete Medical Staff", "description": "Remove medical staff members", "category": "medical", "resource": "staff", "action": "delete"},
    
    # Patient Management
    {"id": "patient_create", "name": "Create Patients", "description": "Register new patients", "category": "patient", "resource": "patients", "action": "create"},
    {"id": "patient_read", "name": "View Patients", "description": "Access patient information", "category": "patient", "resource": "patients", "action": "read"},
    {"id": "patient_update", "name": "Update Patients", "description": "Modify patient records", "category": "patient", "resource": "patients", "action": "update"},
    {"id": "patient_delete", "name": "Delete Patients", "description": "Remove patient records", "category": "patient", "resource": "patients", "action": "delete"},
    {"id": "patient_medical_history", "name": "Access Medical History", "description": "View patient medical history", "category": "patient", "resource": "medical_history", "action": "read"},
    
    # Screening Management
    {"id": "screening_create", "name": "Create Screenings", "description": "Create new screening sessions", "category": "screening", "resource": "screenings", "action": "create"},
    {"id": "screening_read", "name": "View Screenings", "description": "Access screening data and results", "category": "screening", "resource": "screenings", "action": "read"},
    {"id": "screening_update", "name": "Update Screenings", "description": "Modify screening information", "category": "screening", "resource": "screenings", "action": "update"},
    {"id": "screening_delete", "name": "Delete Screenings", "description": "Remove screening records", "category": "screening", "resource": "screenings", "action": "delete"},
    {"id": "screening_results_manage", "name": "Manage Screening Results", "description": "Input and modify screening results", "category": "screening", "resource": "screening_results", "action": "manage"},
    
    # Screening Types
    {"id": "screening_type_comprehensive", "name": "Comprehensive Screening", "description": "Access comprehensive eye screening", "category": "screening", "resource": "screening_types", "action": "comprehensive"},
    {"id": "screening_type_distance", "name": "Distance Vision Screening", "description": "Access distance vision screening", "category": "screening", "resource": "screening_types", "action": "distance"},
    {"id": "screening_type_near", "name": "Near Vision Screening", "description": "Access near vision screening", "category": "screening", "resource": "screening_types", "action": "near"},
    {"id": "screening_type_color", "name": "Color Vision Screening", "description": "Access color vision screening", "category": "screening", "resource": "screening_types", "action": "color"},
    {"id": "screening_type_depth", "name": "Depth Perception Screening", "description": "Access depth perception screening", "category": "screening", "resource": "screening_types", "action": "depth"},
    {"id": "screening_type_binocular", "name": "Binocular Vision Screening", "description": "Access binocular vision screening", "category": "screening", "resource": "screening_types", "action": "binocular"},
    {"id": "screening_type_diagnostic", "name": "Diagnostic Screening", "description": "Access diagnostic screening types", "category": "screening", "resource": "screening_types", "action": "diagnostic"},
    {"id": "screening_type_mobile", "name": "Mobile Unit Screening", "description": "Access mobile unit screening", "category": "screening", "resource": "screening_types", "action": "mobile"},
    {"id": "screening_type_standard", "name": "Standard Vision Screening", "description": "Access standard vision screening", "category": "screening", "resource": "screening_types", "action": "standard"},
    {"id": "screening_type_enhanced", "name": "Enhanced Screening", "description": "Access enhanced screening options", "category": "screening", "resource": "screening_types", "action": "enhanced"},
    
    # School Management
    {"id": "school_create", "name": "Create Schools", "description": "Register new schools", "category": "school", "resource": "schools", "action": "create"},
    {"id": "school_read", "name": "View Schools", "description": "Access school information", "category": "school", "resource": "schools", "action": "read"},
    {"id": "school_update", "name": "Update Schools", "description": "Modify school details", "category": "school", "resource": "schools", "action": "update"},
    {"id": "school_delete", "name": "Delete Schools", "description": "Remove school records", "category": "school", "resource": "schools", "action": "delete"},
    {"id": "school_students_manage", "name": "Manage School Students", "description": "Manage students within schools", "category": "school", "resource": "students", "action": "manage"},
    {"id": "school_teachers_manage", "name": "Manage School Teachers", "description": "Manage teachers within schools", "category": "school", "resource": "teachers", "action": "manage"},
    {"id": "school_screenings_manage", "name": "Manage School Screenings", "description": "Organize screenings for schools", "category": "school", "resource": "school_screenings", "action": "manage"},
    
    # Inventory Management
    {"id": "inventory_create", "name": "Create Inventory Items", "description": "Add new inventory items", "category": "inventory", "resource": "inventory", "action": "create"},
    {"id": "inventory_read", "name": "View Inventory", "description": "Access inventory information", "category": "inventory", "resource": "inventory", "action": "read"},
    {"id": "inventory_update", "name": "Update Inventory", "description": "Modify inventory items", "category": "inventory", "resource": "inventory", "action": "update"},
    {"id": "inventory_delete", "name": "Delete Inventory Items", "description": "Remove inventory items", "category": "inventory", "resource": "inventory", "action": "delete"},
    {"id": "inventory_glasses_manage", "name": "Manage Glasses Inventory", "description": "Manage glasses and frames inventory", "category": "inventory", "resource": "glasses", "action": "manage"},
    
    # Reporting
    {"id": "reports_create", "name": "Create Reports", "description": "Generate new reports", "category": "reporting", "resource": "reports", "action": "create"},
    {"id": "reports_read", "name": "View Reports", "description": "Access existing reports", "category": "reporting", "resource": "reports", "action": "read"},
    {"id": "reports_export", "name": "Export Reports", "description": "Export reports in various formats", "category": "reporting", "resource": "reports", "action": "export"},
    {"id": "reports_analytics", "name": "Access Analytics", "description": "View analytics and insights", "category": "reporting", "resource": "analytics", "action": "read"},
    {"id": "reports_ai_insights", "name": "AI Insights Access", "description": "Access AI-generated insights and recommendations", "category": "reporting", "resource": "ai_insights", "action": "read"},
    
    # Appointments
    {"id": "appointments_create", "name": "Create Appointments", "description": "Schedule new appointments", "category": "appointment", "resource": "appointments", "action": "create"},
    {"id": "appointments_read", "name": "View Appointments", "description": "Access appointment schedules", "category": "appointment", "resource": "appointments", "action": "read"},
    {"id": "appointments_update", "name": "Update Appointments", "description": "Modify appointment details", "category": "appointment", "resource": "appointments", "action": "update"},
    {"id": "appointments_delete", "name": "Cancel Appointments", "description": "Cancel or remove appointments", "category": "appointment", "resource": "appointments", "action": "delete"},
    
    # LINE Bot Management
    {"id": "linebot_settings_manage", "name": "Manage LINE Bot Settings", "description": "Configure LINE Bot settings and features", "category": "integration", "resource": "linebot", "action": "manage"},
    {"id": "linebot_messages_send", "name": "Send LINE Bot Messages", "description": "Send messages through LINE Bot", "category": "integration", "resource": "linebot_messages", "action": "send"},
    {"id": "linebot_users_manage", "name": "Manage LINE Bot Users", "description": "Manage LINE Bot user interactions", "category": "integration", "resource": "linebot_users", "action": "manage"},
    
    # Panel Settings
    {"id": "panel_settings_view", "name": "View Panel Settings", "description": "Access panel configuration settings", "category": "system", "resource": "panel_settings", "action": "read"},
    {"id": "panel_settings_update", "name": "Update Panel Settings", "description": "Modify panel configuration", "category": "system", "resource": "panel_settings", "action": "update"},
    
    # File Management
    {"id": "files_upload", "name": "Upload Files", "description": "Upload files to the system", "category": "system", "resource": "files", "action": "upload"},
    {"id": "files_download", "name": "Download Files", "description": "Download files from the system", "category": "system", "resource": "files", "action": "download"},
    {"id": "files_delete", "name": "Delete Files", "description": "Remove files from the system", "category": "system", "resource": "files", "action": "delete"},
]
